cli/internal/publish/pluginui.go

```diff
@@ -65,14 +81,35 @@ func UploadPluginUIAssets(ctx context.Context, c *cloudquery_api.ClientWithRespo
 	eg, egCtx := errgroup.WithContext(ctx)
 	eg.SetLimit(4)
 
+	var bundleAsset cloudquery_api.PluginUIAsset
 	for _, asset := range resp.JSON201.Assets {
+		if asset.Name == uiAssetBundleTarName {
+			bundleAsset = asset
+			continue
+		}
+
 		asset := asset
 		details := urlPathVsDetails[asset.Name]
 		eg.Go(func() error {
 			return hub.UploadFileWithContentType(egCtx, asset.UploadURL, details[0], details[1])
 		})
 	}
 
+	if bundleAsset.UploadURL == "" {
+		return errors.New("bundle asset URL not found in the response")
+	}
+
+	bundleReader, bundleWriter := io.Pipe()
+	defer bundleWriter.Close()
```

Comment:
Should `bundleReader` be closed as well?

Reply:
We do close the file here https://github.com/cloudquery/cloudquery/pull/20116/files#diff-9151361555ef4493db87b7c5d100db9d6666f39a38245b40957be1a63d219a53L79 before calling the `io.Reader` variant of `UploadReaderWithContentType`

Reply:
I'm pretty sure `io.PipeReader` is safe to just discard (no internal references and no held resources) but [5dffd31](https://github.com/cloudquery/cloudquery/pull/20116/commits/5dffd3178590504bbe8b414e3cbde661fc7372c1) adds an explicit close.